syntax = "proto3";

package inventory;

message CheckAvailabilityRequest {
    int32 product_id = 1;
    int32 required_qty = 2;
}

message CheckAvailabilityResponse {
    bool available = 1;
}

message UpdateInventoryRequest {
    int32 product_id = 1;
    int32 change_qty = 2;
}

message UpdateInventoryResponse {
    bool success = 1;
}

service InventoryService {
    rpc CheckAvailability (CheckAvailabilityRequest) returns (CheckAvailabilityResponse);
    rpc UpdateInventory (UpdateInventoryRequest) returns (UpdateInventoryResponse);
}
